janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geoshape.java

```diff
@@ -547,6 +548,16 @@ public Geoshape read(ScanBuffer buffer) {
             try {
                 return GeoshapeBinarySerializer.read(inputStream);
             } catch (IOException e) {
+                if (buffer instanceof ReadArrayBuffer && l == 1) {
+                    // retry using legacy point deserialization
+                    ((ReadArrayBuffer) buffer).movePositionTo(0);
+                    VariableLong.readPositive(buffer);
+                    try {
+                        return point(buffer.getFloat(), buffer.getFloat());
+                    } catch (Exception e2) {
+                        // throw original exception
+                    }
+                }
                 throw new RuntimeException("I/O exception reading geoshape");
```

Comment:
say if (!(buffer instances ReadArrayBuffer) || l != 1) throw new RuntimeException("I/O exception reading geoshape", e)
then when you catch (Exception e2), you can throw a more specific exception like "unable to create a point from buffer"